src/hardware/vga_draw.cpp

```diff
@@ -1614,23 +1614,27 @@ void VGA_SetupDrawing(uint32_t /*val*/)
 	case M_HERC_GFX:
 		vga.draw.blocks=width*2;
 		width*=16;
-		aspect_ratio = (static_cast<float>(width) / static_cast<float>(height)) * (3.0f / 4.0f);
+		assert(width > 0);
+		assert(height > 0);
+		aspect_ratio = static_cast<float>(width * 3) / static_cast<float>(height * 4);
 		VGA_DrawLine = VGA_Draw_1BPP_Line;
 		break;
 	case M_TANDY2:
 		aspect_ratio = 1.2f;
 		doubleheight=true;
-		if (machine==MCH_PCJR) doublewidth=(vga.tandy.gfx_control & 0x8)==0x00;
-		else doublewidth=(vga.tandy.mode_control & 0x10)==0;
+		doublewidth = machine == MCH_PCJR
+		                      ? !(vga.tandy.gfx_control & 0x8)
+		                      : !(vga.tandy.mode_control & 0x10);
 		vga.draw.blocks = width * (doublewidth ? 1 : 2);
 		width = vga.draw.blocks * 8;
 		VGA_DrawLine = VGA_Draw_1BPP_Line;
 		break;
 	case M_TANDY4:
 		aspect_ratio = 1.2f;
 		doubleheight=true;
-		if (machine==MCH_TANDY) doublewidth=(vga.tandy.mode_control & 0x10)==0;
-		else doublewidth=(vga.tandy.mode_control & 0x01)==0x00;
+		doublewidth = machine == MCH_TANDY
+		                      ? !(vga.tandy.mode_control & 0x10)
```

Comment:
You did not address the actual problem reported by Coverity. This line should be `mode_control` or `gfx_control`? It does indeed look like a copy-paste error right now (still).

Reply:
> You did not address the actual problem reported by Coverity.

Coverity is doing a simple pattern-comparison and assuming that `gfx_control` should be used in both of the TANDY_2 case conditions, and that `mode_control` should be used both TANDY_4 case conditions:

![2021-08-07_14-46](https://user-images.githubusercontent.com/1557255/128614708-19eca2b9-d884-48f8-9a6e-0f9dff4f80f1.png)

However, I'm reluctant to change the functionality without having games to test with (Seven Cities of Gold and BC Quest for Tires get into the TANDY_4 else branch, but not TANDY_2 else branch).

Reply:
I've added a TODO as a comment in the code. 
(Turns out this PR was probably useless.. sigh).